mlx/ops.cpp

```diff
@@ -83,10 +83,24 @@ array arange(
   if (std::isnan(start) || std::isnan(step) || std::isnan(stop)) {
     throw std::invalid_argument("[arange] Cannot compute length.");
   }
+
+  if (std::isinf(start) || std::isinf(stop)) {
+    throw std::invalid_argument("[arange] Cannot compute length.");
+  }
+
+  // Check if start and stop specify a valid range because if not, we have to
+  // return an empty array
+  if (std::isinf(step) &&
+      (step > 0 && start < stop || step < 0 && start > stop)) {
+    return array({start}, dtype);
+  }
+
   double real_size = std::ceil((stop - start) / step);
-  if (std::isnan(real_size)) {
+
+  if (real_size < INT_MIN || real_size > INT_MAX) {
     throw std::invalid_argument("[arange] Cannot compute length.");
   }
```

Comment:
```suggestion
  if (real_size > INT_MAX) {
    throw std::invalid_argument("[arange] Maximum size exceeded.");
  }
```

Shouldn't we let the `std::max` below handle the negative cases?

Reply:
Yes, you're correct.